File: StructureTest1/results/StructureTest1.cpp
#include <one.hpp>
template<typename T>
class List {
  public:
    vec<T> items;

  private:
};

class Item {
  public:
    int offset = 5;
    string str_test = string("test") + string("test2");
    string str_constr = string("constr");

    Item(string str_constr) {
        this->str_constr = str_constr;
    }

  private:
};

class Container {
  public:
    sp<List<Item>> item_list;
    sp<List<string>> string_list;

    void method0() {
    }
    
    string method1(string str) {
        return str;
    }

  private:
};

File: CompilationTest/StructureTest1/results/StructureTest1.rb
class MyList
  attr_accessor(:items)
end

class Item
  attr_accessor(:offset)
  attr_accessor(:str_test)
  attr_accessor(:str_constr)

  def initialize(str_constr)
      @offset = 5
      @str_test = "test" + "test2"
      @str_constr = "constr"

      self.str_constr = str_constr
  end
end

class Container
  attr_accessor(:item_list)
  attr_accessor(:string_list)

  def method0()
  end

  def method1(str)
      return str
  end
end

puts "ok"